tests/model/ManyManyListTest.php

```diff
@@ -306,7 +307,41 @@ public function testFilteringOnPreviouslyJoinedTable() {
 		$this->assertEquals(1, $productsRelatedToProductB->count());
 	}
 
-
+    public function testWriteManipulationWithNonScalarValuesAllowed()
+    {
+        $left = MockDynamicAssignmentDataObject::create();
+        $left->write();
+        $right = MockDynamicAssignmentDataObject::create();
+        $right->write();
+        $left->MockManyMany()->add($right, array(
+            'ManyManyStaticScalarOnlyField' => true,
+            'ManyManyDynamicScalarOnlyField' => false,
+            'ManyManyDynamicField' => true
+        ));
+        $pivot = $left->MockManyMany()->first();
+        $this->assertEquals(1, $pivot->ManyManyStaticScalarOnlyField);
+        $this->assertEquals(0, $pivot->ManyManyDynamicScalarOnlyField);
+        $this->assertEquals(1, $pivot->ManyManyDynamicField);
+    }
+
+    public function testWriteManipulationWithNonScalarValuesDisallowed()
+    {
+        $left = MockDynamicAssignmentDataObject::create();
+        $left->write();
+        $right = MockDynamicAssignmentDataObject::create();
+        $right->write();
+
+        try {
+            $left->MockManyMany()->add($right, array(
+                'ManyManyStaticScalarOnlyField' => true,
+                'ManyManyDynamicScalarOnlyField' => false,
+                'ManyManyDynamicField' => true
+            ));
+            $this->fail("Expected exception \"parameterised field assignments are disallowed\"");
```

Comment:
You can use `/** @expectedExceptionMessageRegExp /regex match on exception message/ */` if you want to

Reply:
what's the difference between these two tests?

Reply:
Fix the test to use annotation and to actually generate an error.